6.5 release notes

pdurbin opened the pull request:
**What this PR does / why we need it**:

Release notes, as usual. Preview at https://github.com/IQSS/dataverse/blob/10952-release-notes/doc/release-notes/6.5-release-notes.md

**Which issue(s) this PR closes**:

- Closes #10952

**Special notes for your reviewer**:

I'll leave some comments inline.

**Suggestions on how to test this**:

Go through the upgrade steps on demo. Here's the issue for that:

- #10953

pdurbin commented on doc/release-notes/6.5-release-notes.md:
````diff
@@ -0,0 +1,367 @@
+service payara start
+```
+
+6\. Update metadata blocks
+
+No changes required for this release.
+
+7\. Update Solr schema.xml file. Start with the standard v6.4 schema.xml, then, if your installation uses any custom or experimental metadata blocks, update it to include the extra fields (step 7a).
````
We should remove 6 and shift 7 up. As discussed with @ofahimIQSS, we plan to copy and paste some improved Solr upgrade steps from the snippet in this PR, assuming it passes QA and makes it into 6.5:

- #10887

pdurbin replied:
I renumbered these and added the Solr steps (sometime last week). Resolving.

pdurbin commented on doc/release-notes/6.5-release-notes.md:
```diff
@@ -0,0 +1,367 @@
+- UX improvements, especially Preview URLs
+- increased harvesting flexibility
+- performance gains
+- a security vulnerability addressed
```
I don't think we ever put our a security announcement. We probably should. See https://guides.dataverse.org/en/6.4/developers/security.html#sending-security-notices

ofahimIQSS commented on doc/release-notes/6.5-release-notes.md:
```diff
@@ -0,0 +1,367 @@
+- increased harvesting flexibility
+- performance gains
+- a security vulnerability addressed
+- many bugs fixes
```
```suggestion
- many bug fixes
```

ofahimIQSS commented on doc/release-notes/6.5-release-notes.md:
```diff
@@ -0,0 +1,367 @@
+
+### More Scalable Dataset Version Differencing
+
+Differencing between dataset versions, which is done during dataset edit operations and to populate the dataset page versions table has been made signficantly more scalable. See #10814 and #10818.
```
```suggestion
Differencing between dataset versions, which is done during dataset edit operations and to populate the dataset page versions table has been made significantly more scalable. See #10814 and #10818.
```

pdurbin commented on doc/release-notes/6.5-release-notes.md:
````diff
@@ -0,0 +1,367 @@
+In the following commands, we assume that Payara 6 is installed in `/usr/local/payara6`. If not, adjust as needed.
+
+```shell
+export PAYARA=/usr/local/payara6`
````
```suggestion
export PAYARA=/usr/local/payara6
```

sekmiller commented on doc/release-notes/6.5-release-notes.md:
```diff
@@ -0,0 +1,385 @@
+
+## Settings Added
+
+- `:GlobusBatchLookupSize`
```
Do we need a description/explanation here?

pdurbin replied:
We've always had a "settings added" section and I think it's helpful to the sysadmins to always have this list to go to, even if it's short this time.

ofahimIQSS commented on doc/release-notes/6.5-release-notes.md:
```diff
@@ -0,0 +1,385 @@
+
+### Relation Type (Related Publication) and DataCite
+
+The subfield "Relation Type" was added to the field "Related Publication" in Dataverse 6.4 (#10632) but couldn't be used without workarounds described in an [announcement](https://groups.google.com/g/dataverse-community/c/zlRGJtu3x4g/m/GtVZ26uaBQAJ) about the problem. The bug has been fixed and workarounds and are longer required. See #10926 and the announcement above.
```
```suggestion
The subfield "Relation Type" was added to the field "Related Publication" in Dataverse 6.4 (#10632) but couldn't be used without workarounds described in an [announcement](https://groups.google.com/g/dataverse-community/c/zlRGJtu3x4g/m/GtVZ26uaBQAJ) about the problem. The bug has been fixed and workarounds are no longer required. See #10926 and the announcement above.
```

sekmiller commented:
Looks good overall. I left a couple of comments/questions

ofahimIQSS commented on doc/release-notes/6.5-release-notes.md:
```diff
@@ -0,0 +1,385 @@
+
+### Updating Files Now Possible When Latest and Only Dataset Version is Deaccessioned
+
+When a dataset was deaccessioned and was the only previous version it would cause an error when trying to update the files. This has been fixed. See #9351 and #10901.
```
```suggestion
When a dataset was deaccessioned, and was the only previous version, it would cause an error when trying to update the files. This has been fixed. See #9351 and #10901.
```

pdurbin commented on doc/release-notes/6.5-release-notes.md:
```diff
@@ -0,0 +1,385 @@
+
+Replace schema.xml
+
+Please note that the path to Solr may differ than the example below.
```
Hmm, the main audience is not developers so I don't think so. Maybe we should sent and email to the dev list?

ofahimIQSS commented on doc/release-notes/6.5-release-notes.md:
```diff
@@ -0,0 +1,385 @@
+
+Replace schema.xml
+
+Please note that the path to Solr may differ than the example below.
```
```suggestion
Please note that the path to Solr may differ from the example below.
```

qqmyers reviewed (approved):
Looks good. The one nit I have is that using the api/admin/index call (step 8) is not actually the one we talk about as reindex in place in the guides. That refers to deleting the time stamps for indexing and then calling continue. We could ignore this, just drop the 'reindex-in-place' phrase, or show the other calls. I'm not sure which is best - if the server is restarted before the /api/admin/index call completes, is there a way to restart like there is if you clear the index or delete the timestamps?

pdurbin commented:
@qqmyers good catch. In 2de57e1 I removed "in place".

In other news, I saw the cvoc PR got merged so in fe13a43 I removed the snippet and worked it into the release notes.

ofahimIQSS commented:
reviewed and proof-read. Merging PR
